save-agent/src/nativeMain/kotlin/org/cqfn/save/agent/SaveAgent.kt

```diff
@@ -128,18 +139,25 @@ class SaveAgent(private val config: AgentConfiguration,
             val result = runCatching {
                 postExecutionData(executionData)
             }
-            if (result.isSuccess) {
+            if (result.isSuccess && result.getOrNull()?.statusCode == HttpStatusCode.OK) {
                 return@repeat
```

Comment:
Why do we return at repeat here?

Reply:
I guess I initially planned to add some logic after `repeat`, but now it has no difference from usual `return`. I'll change.

Reply:
Actually, `repeat` is a non-inlined function and therefore plain `return` is forbidden there. So I added it back :)